include/exadg/grid/grid_utilities.h

```diff
@@ -245,6 +246,53 @@ create_fine_and_coarse_triangulations(
   }
 }
 
+/**
+ * This functions reads an external triangulation defined in the file "file_in" and stores it in
+ * the triangulation "tria".
+ */
+template<int dim>
+inline void
+read_external_triangulation(dealii::Triangulation<dim, dim> & tria, GridData const & data)
+{
+  AssertThrow(!data.file_name.empty(),
+              dealii::ExcMessage(
+                "You are trying to read a grid file, but the string, which is supposed to contain"
+                " the file, is empty. Most likely, you forgot to specify the file location in the"
+                " input file. The input file of the Poisson/sine application can be an example"
+                " showing how to set it correctly."));
+
+  dealii::GridIn<dim> grid_in;
+
+  grid_in.attach_triangulation(tria);
+
+  // find the file extension from the given file_in string
+  std::string extension = data.file_name.substr(data.file_name.find_last_of('.') + 1);
+
+  AssertThrow(!extension.empty(),
+              dealii::ExcMessage("You are trying to read a grid file, but the file extension is"
+                                 " empty. Check the input file to make sure that the file"
+                                 " extension is correctly defined after a full stop."));
+
+  // decide the file format
+  typename dealii::GridIn<dim>::Format format;
+  if(extension == "e" || extension == "exo")
+    format = dealii::GridIn<dim>::Format::exodusii;
+  else
+    format = grid_in.parse_format(extension);
+
+  // TODO: check if the exodusIIData is needed
+  // typename dealii::GridIn<dim>::ExodusIIData exodusIIData;
+
+  grid_in.read(data.file_name, format);
+
+  AssertThrow(get_element_type(tria) == data.element_type,
+              dealii::ExcMessage("You are trying to read a grid file, but the element type of the"
+                                 " external grid file and the element type specified in the"
+                                 " application don't match. Most likely, you forgot to change the"
```

Comment:
The comment "specified in the application" is not precise enough. There is no need to set this parameter in the application file, since this parameter has a default value. Since we cannot know how GridData is used and will be used in the future, we should not refer to "application" here.